CONTRIBUTING.md

```diff
@@ -5,3 +5,71 @@ Thanks for considering contributing to Composer!
 Issues tagged with [good first issue](https://github.com/mosaicml/composer/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) are great options to start contributing.
 
 If you have questions, join us on [Slack](https://join.slack.com/t/mosaicml-community/shared_invite/zt-w0tiddn9-WGTlRpfjcO9J5jyrMub1dg) -- we'll be happy to help you!
+
+We welcome contributions for bug fixes, new efficient methods you'd like to contribute to the community, or new models and datasets!
+
+## New Algorithms
+
+Have a new algorithm you'd like to contribute to the library as part of your research? Reach out to us on Slack or file a new issue with the proposed method, and we'll work with you to get started!
```

Comment:
What if people just built it for themselves and want to open a PR directly? "we'll work with you to get started" sounds 1) it's hard and 2) there's a hurdle to jump through, sort of like when a pricing page says "get in touch" instead of a price.

We can always reject PRs that aren't up to our standards, so I'd vote for saying new method PRs are welcome, but we suggest opening an issue first to discuss how and whether it makes sense to do it.